### PR TITLE
Remove old FSDP1 `fully_shard`

### PR DESCRIPTION
**Pull request by awgu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #141868
* __->__ #141875

FSDP1's `fully_shard` frontend was an exploration at the end of 2022 H2 as part of the `torch/distributed/_composable` APIs to avoid `nn.Module` wrappers. It calls into the same backend code as FSDP1's `FullyShardedDataParallel`.

The API did not gain traction internally, so we instead reused the name `fully_shard` for FSDP2, which similarly is not an `nn.Module` wrapper and follows similar design principles as FSDP1's `fully_shard`.

To the best of our knowledge, we have removed all instances of FSDP1's `fully_shard` internally, and we put the deprecation warning in open source in 2.4 saying it will be removed after 2.5 (which is now):
https://github.com/pytorch/pytorch/blob/4959784dac633cc23fc59401eb8cfcae47a07f4c/torch/distributed/_composable/fully_shard.py#L40-L48

We are skipping the PR sanity check because this PR is only removing code, not adding new code, and should not require this sanity check.

cc @H-Huang @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @albanD @LucasLLC @MeetVadakkanchery @mhorowitz @pradeepfn

Differential Revision: [D66664988](https://our.internmc.facebook.com/intern/diff/D66664988)

### CODE REVIEW

File: test/distributed/checkpoint/test_state_dict.py [hunk 1]
```diff
@@ -165,6 +159,8 @@ def _test_fsdp(
         compile_model: bool = False,
         optimizer_class: Type[Optimizer],
     ) -> None:
+        if use_composable:
+            raise AssertionError("Composable FSDP1 is no longer supported")
```
Comment (awgu) [inline test/distributed/checkpoint/test_state_dict.py:163]: delete this and remove the `use_composable` arg altogether

**Comment by fegin** [comment]
Should we import this PR to internal first to verify if this breaks any internal tests even though the use cases are all deprecated?

**Comment by awgu** [comment]
@fegin yes, I will import it once the CI on Github stabilizes. Searching internally, I did not see any uses left after @wz337 helped remove, but I will import just in case.

**Comment by awgu** [comment]
@awgu has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D66664988).

**Comment by awgu** [comment]
@awgu has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D66664988).

**Comment by awgu** [comment]
@awgu has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D66664988).

**Comment by awgu** [comment]
@awgu has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D66664988).

**Review by weifengpy — APPROVED** [review]

**Pull request closed without merging**